core/services/vrf/listener_v2.go

```diff
@@ -460,17 +460,24 @@ func (lsn *listenerV2) shouldProcessSub(subID uint64, sub vrf_coordinator_v2.Get
 
 	gasPriceWei := lsn.cfg.KeySpecificMaxGasPriceWei(fromAddress)
 
-	estimatedFee, err := lsn.estimateFeeJuels(vrfRequest, gasPriceWei)
-	if err != nil {
-		l.Warnw("Couldn't estimate fee, processing sub anyway", "err", err)
-		return true
-	}
+	simResult := lsn.simulateFulfillment(ctx, gasPriceWei, reqs[0], l)
```

Comment:
One tricky thing here is we don't check if the request has been fulfilled. Should we loop through `reqs` to find the first one that hasn't been fulfilled and simulate that instead? 

Otherwise, I'm wondering if we could get a simulation failure purely based on the request already being fulfilled on chain. However, it seems that if that is the case, a commitment would no longer be on-chain in that case, and in processRequestsPerSub we would mark that as "already fulfilled" and won't try fulfilling it again.